source/localizable/tutorial/location-service.md

```diff
@@ -0,0 +1,301 @@
+
+### Making Google Maps Available in Ember
+
+There are several ways to include 3rd party libraries in Ember. See the guides section on [managing dependencies](../../addons-and-dependencies/managing-dependencies/) as a starting point when you need to add one.
+
+Since Google provides its map API as a remote script, we'll use curl to download it into our project's vendor directory.
+
+```shell
+curl -o vendor/gmaps.js https://maps.googleapis.com/maps/api/js?v=3.22
+```
+Once in the vendor directory, the script can be built into our app.
+We'll just need to import it in our build file.
+
+```ember-cli-build.js
+/*jshint node:true*/
+/* global require, module */
+var EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function(defaults) {
+  var app = new EmberApp(defaults, {
+    // Add options here
+  });
+
+  // Use `app.import` to add additional libraries to the generated
+  // output files.
+  //
+  // If you need to use different assets in different
+  // environments, specify an object as the first parameter. That
+  // object's keys should be the environment name and the values
+  // should be the asset to use in that environment.
+  //
+  // If the library that you are including contains AMD or ES6
+  // modules that you would like to import into your application
+  // please specify an object with the list of modules as keys
+  // along with the exports of each module as its value.
+  app.import('vendor/gmaps.js');
+
+  return app.toTree();
+};
+```
+
+### Loading Maps into Application Code
+
+Now that the google map API library is built into our app, lets create a service to wrap them.
+Services are used to package common concerns and keep shared state across the application
+
+Accessing our maps API through a service will give us several benefits
+
+* It is injected with a service locator pattern,
+  meaning it will abstract the maps API from the code that uses it, allowing for easier refacoring and maintenance
+* It is lazy-loaded, meaning it won't be initialized until its called the first time.
+* It is a singleton, which could allow for potential caching of map data.
+* It follows a lifecycle, meaning we have hooks to execute cleanup code when the app stopped.
+
+Lets get started creating our service by generating it though Ember CLI.
+
+```shell
+ember g service maps
+```
+The service provides functions to create the map, geocode an address, and create/render directions.
+Since the google api makes asynchronous calls for geoding and directions rendering,
+the service will use Ember's promise API so that the caller can handle the response locally, instead of passing callbacks around.
+
+```app/services/maps.js
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+
+  init() {
+    this._super(...arguments);
+    this.set('directionsService', new google.maps.DirectionsService());
+    this.set('geocoder', new google.maps.Geocoder());
```

Comment:
Hmm, we're using a global here?  Seems like it might be better to use a shim in our vendors folder and import that instead, no?  An example of what I'm discussing is here: https://www.foraker.com/blog/google-spreadsheets-on-ember (just used it yesterday, so it's fresh ;-) )